frontend/src/components/AuthForm.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="auth-form">
+    <h2>{{ isLogin ? 'Login' : 'Register' }}</h2>
+    <form @submit.prevent="handleSubmit">
+      <input 
+        v-if="!isLogin" 
+        v-model="username" 
+        placeholder="Username" 
+        required 
+      />
+      <input 
+        v-model="email" 
+        type="email" 
+        placeholder="Email" 
+        required 
+      />
+      <input 
+        v-model="password" 
+        type="password" 
+        placeholder="Password" 
+        required 
+      />
+      <button type="submit">{{ isLogin ? 'Login' : 'Register' }}</button>
+    </form>
+    <p v-if="message">{{ message }}</p>
+    <button @click="toggleMode">
+      <router-link :to="isLogin ? '/register' : '/login'">
+        {{ isLogin ? 'Need an account? Register' : 'Already have an account? Login' }}
+      </router-link>
+    </button>
+  </div>
+</template>
+
+<script>
+import authService from '@/services/authService'; 
+
+export default {
+  props: {
+    isLogin: {
+      type: Boolean,
+      default: true
+    }
+  },
+  data() {
+    return {
+      username: '',
+      email: '',
+      password: '',
+      message: ''
+    };
+  },
+methods: {
+  async handleSubmit() {
+    try {
+      let response;
+      if (this.isLogin) {
+        const payload = {
```

Comment:
Znam da su ti varijable već označene kao required, ali nije loše i ovde uraditi validation check pre nego što pošalješ payload.